src/components/personoppgave/BehandlePersonOppgaveKnapp.tsx

```diff
@@ -21,13 +22,19 @@ const getFerdigbehandletPrefixText = (personoppgaveType: PersonOppgaveType) => {
   }
 };
 
-const getFerdigbehandletText = (personOppgave: PersonOppgave) => {
+const getFerdigbehandletText = (
+  personOppgave: PersonOppgave,
+  veilederNavn: string | undefined
```

Comment:
Lurer på om dette er det samme? Cleanere måte å skrive det på kanskje?😊
```suggestion
  veilederNavn?: string
```

Reply:
Spørs om propertyen faktisk er optional eller om den er påkrevd, men kan være undefined. Kan være viktig å skille på det ref https://nikolasbarwicki.com/articles/optional-vs-undefined-typescript-required-properties

Reply:
Nja litt usikker - jeg tror ikke vi får noen syntaksfeil i koden ved å endre til dette, men det handler kanskje mer om hvilket API vi ønsker at funksjonen skal ha.

Når vi sier `veilederNavn: string | undefined` så sier vi at `veilederNavn` _må_ sendes med, men at verdien vi sender kan være undefined. 
Når vi sier `veilederNavn?: string` så tenker jeg at det er valgfritt å sende med `veilederNavn`, men i de tilfellene vi faktisk sender en verdi så vil jeg forvente at den er `string` (selv om det tillates å være undefined, så man må jo håndtere det i koden uansett).

Akkurat her føler jeg vi vil kreve å sende med `veilederNavn` for de som skal bruke funksjonen, og derfor er parameteret påkrevd.

Det er hvertfall min forståelse av forskjellen, men vet ikke om noen andre har bedre peiling og kan utdype?

Reply:
Synes det var en god oppsummering, og det virker riktigst å bruke `veilederNavn: string | undefined` i dette tilfellet 👍